content/250-postgres/300-database/750-serverless-driver.mdx

```diff
@@ -0,0 +1,177 @@
+| :----------------- | :------------ | :------- | :-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `connectionString` | `string`      | **Yes**  | The connection string for your Prisma Postgres instance.                                                                                                                                      |
+| `deserialize`      | `Deserialize` | No       | Custom deserializer function which takes a column type OID and a raw value and returns a mapped value. It's type is defined as `type Deserialize = (value: unknown, oid: unknown) => unknown` |
+
+#### Usage
+
+##### With manual type definitions
+
+```ts
+import { ppg } from "@prisma/ppg";
+
+type Post = {
+  id: number;
+  title: string;
+  content: string | null;
+  published: boolean;
+  authorId: number | null;
+}
+
+type User = {
+  id: number;
+  email: string
+}
+
+const sql = ppg("prisma+postgres://accelerate.prisma-data.net/?api_key=...");
+
+const posts: Post[] = await sql<Post>`SELECT * FROM "Post"`
+
+const userId = 42;
+const user: User[] = await sql<User>`SELECT * FROM "User" WHERE "id" = ${userId}`
+```
+
+##### With generated Prisma Client types
+
+```ts
+import { ppg } from "@prisma/ppg";
+import type { User, Post } from "@prisma/client";
+
+const sql = ppg("prisma+postgres://accelerate.prisma-data.net/?api_key=...");
+
+const posts: Post[] = await sql<Post>`SELECT * FROM "Post"`
+
+const userId = 42;
+const user: User[] = await sql<User>`SELECT * FROM "User" WHERE "id" = ${userId}`
+```
+
+
+### `Client`
+
+The `Client` class provides more low-level control and should be used with more care. It aims to be a drop-in replacement for the [`pg`](https://node-postgres.com/) driver:
+
+```ts
+class Client implements Queryable {
+    constructor(options: ClientOptions);
+
+    /**
+     * Executes a query against the Prisma Postgres database.
+     */
+    query(query: string, parameters: unknown[]): Promise<QueryResponse>;
+}
+```
+
+The `query` function it exposes:
+
+- Takes a raw string and a list of params separately, allowing you to control the SQL parameters yourself or concatenate the query unsafely if needed.
+- Returns column types and raw data without converting the rows into an array of objects.
+
+#### Usage
+
+```ts
+import { Client } from "@prisma/ppg";
+
+const client = new Client({
+  connectionString: "prisma+postgres://accelerate.prisma-data.net/?api_key=...",
+});
+
+const posts = await client.query('SELECT * FROM "Post" WHERE "authorId" = $1', [1]);
+```
+
+This query returns an object of this structure:
+
+```ts
+{
+  columns: [
+    { name: 'id', oid: 23, type: 23 },
+    { name: 'title', oid: 25, type: 25 },
+    { name: 'content', oid: 25, type: 25 },
+    { name: 'published', oid: 16, type: 16 },
+    { name: 'authorId', oid: 23, type: 23 }
```

Comment:
The `type` field is not defined in the types and users shouldn't rely on it, only on `oid`. We should remove this field.